### DISCUSSION TITLE
RTU examples invert the TX and RX pins.

### DISCUSSION THREAD

**Discussion by espivetta** [eModbus/eModbus#327] [General]
There is a problem on the examples of RTU. on my case do the MAX3232 chip freezes and some times burn. 

```c++
// Init Serial2 connected to the RTU Modbus
// (Fill in your data here!)
  RTUutils::prepareHardwareSerial(Serial2);
  Serial2.begin(19200, SERIAL_8N1, GPIO_NUM_17, GPIO_NUM_16);
``` 

The correct sintax for the serial.begin is "Serial2.begin(baud-rate, protocol, RX pin, TX pin)"

The main problem is, the ESP32 hardware is configurated to the GPIO16 be the UART2 RX and the GPIO17 be the UART2 TX. 
![image](https://github.com/eModbus/eModbus/assets/72839020/25dfdad1-8e79-4fdb-bee6-0dadfbe2eec0)

But on the example, the pins will be inverted by the line `Serial2.begin(19200, SERIAL_8N1, GPIO_NUM_17, GPIO_NUM_16);` and this could cause a freeze on MAX3232 when the esp32 resets(cause by a watchdog for example).

Sorry my poor english, but i think managed to get my point of view.

**Comment by Miq1** [comment]
Yes, right, the pins are reversed in regards to the official pinout. The reason is the setup I had with some of my first devices eons ago 😉 
But is is a feature of the ESP32 to let you freely configure the GPIOs for the UARTs, so I would not consider this an error. 
The MAX3232 can not be harmed by that at all, since the logic levels of RX and TX are remaining the same regardless of the GPIOs used - there just will be no data seen if the pins are reversed.
I am afraid your issue seems to be caused by some other reason. If the MAX3232 even can be fried, there must be some really high voltages involved. During reset some of the ESP32's pins may be floating, but normally will not put out more than the regular 3V3 level, so it looks like another component of your setup may be the culprit.
